delta/rdf/src/main/scala/ch/epfl/bluebrain/nexus/delta/rdf/jsonld/decoder/MagnoliaJsonLdDecoder.scala

```diff
@@ -0,0 +1,58 @@
+package ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder
+
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.ExpandedJsonLdCursor
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.decoder.JsonLdDecoderError.DecodingDerivationFailure
+import magnolia.{CaseClass, SealedTrait}
+
+private[decoder] object MagnoliaJsonLdDecoder {
+
+  def combine[A](caseClass: CaseClass[JsonLdDecoder, A])(implicit config: Configuration): JsonLdDecoder[A] = {
+    val keysMap = caseClass.parameters
+      .filter { p => p.label != config.idPredicateName }
+      .map { p =>
+        p.label -> config.context
+          .expand(p.label, useVocab = true)
+          .toRight {
+            val err = s"Label '${p.label}' could not be converted to Iri. Please provide a correct context"
+            DecodingDerivationFailure(err)
+          }
+      }
+      .toMap
+
+    new JsonLdDecoder[A] {
+
+      override def apply(cursor: ExpandedJsonLdCursor): Either[JsonLdDecoderError, A] =
+        caseClass.constructMonadic {
+          case p if p.label == config.idPredicateName => decodeOrDefault(p.typeclass, cursor, p.default)
+          case p                                      =>
+            keysMap(p.label).flatMap { predicate =>
+              val nextCursor = cursor.downField(predicate)
+              decodeOrDefault(p.typeclass, nextCursor, p.default)
+            }
+        }
+
+      private def decodeOrDefault[B](dec: JsonLdDecoder[B], cursor: ExpandedJsonLdCursor, default: Option[B]) =
+        default match {
+          case Some(value) if !cursor.succeeded => Right(value)
+          case _                                => dec(cursor)
+        }
+    }
+  }
+
+  def dispatch[A](sealedTrait: SealedTrait[JsonLdDecoder, A])(implicit config: Configuration): JsonLdDecoder[A] =
+    new JsonLdDecoder[A] {
+
+      override def apply(c: ExpandedJsonLdCursor): Either[JsonLdDecoderError, A] =
+        c.getTypes match {
+          case Right(types) =>
+            sealedTrait.subtypes
```

Comment:
I think it would be useful to be able to configure the values we use to discriminate with a transformation function.
For the IndexingElasticSearchView I need to use the `ElasticSearchView` type value. Would using context be sufficient in this case?

Reply:
I think so, the @context can handle renaming of keys and values.